docs/content/latest/architecture/transactions/read-committed.md

```diff
@@ -0,0 +1,785 @@
+
+1. **SELECTs (without explicit row locking)** -
+    1. New read point is chosen at statement start that includes anything that committed before the query began.
+    2. Data from previous updates from same transaction is visible
+2. **UPDATE, DELETE, SELECT FOR UPDATE, FOR SHARE,** _FOR NO KEY UPDATE, FOR KEY SHARE (the last two are not mentioned in Postgres documentation but the same behavior is seen for these as below.)_
+    1. New read point is chosen at statement start that includes anything that committed before the query began.
+    2. If the row in interest -
+        1. is being updated (or deleted) by other transactions in a conflicting way (i.e., read time of statement falls within read time of other transactions to current time), wait for them to commit/rollback and then perform recheck steps.
+        2. has been updated (or deleted) by other transactions in a conflicting way (i.e., read time of statement falls within read time to commit time of other transactions), perform recheck steps.
+        3. has been locked by other transactions in a conflicting way, wait for them to commit/ rollback and then perform recheck steps.
+        <br />
+        <br />
+
+        **Recheck steps:**
+        1. If a row is deleted, ignore it
+        2. Apply update/ acquire lock on updated version of row if where clause evaluates to true on the updated version of row. (Note that the updated version of a row could have a different pk as well - this implies Postgres follows the chain of updates for a row even across pk changes). 
+3. INSERT
+    1. ON CONFLICT DO UPDATE: if a conflict occurs, wait for the conflicting transaction to commit/ rollback.
+        1. On rollback, proceed as usual
+        2. On commit, modify the new version of row
+    2. ON CONFLICT DO NOTHING: do nothing if a conflict occurs
+
+Apart from the above requirements, there is a YSQL specific requirement: ensure that external clients don’t face kReadRestart errors.
+
+### 3 Usage
+
+Once we set the gflag yb_enable_read_committed_isolation=true, the READ COMMITTED isolation in YSQL will actually map to the READ COMMITTED implementation in docdb. If set to false, it will have the existing behaviour of mapping READ COMMITTED to REPEATABLE READ.
+
+The following ways can be used to start a READ COMMITTED transaction after setting the gflag:
+
+1. `START TRANSACTION isolation level read committed [read write | read only];`
+2. `BEGIN [TRANSACTION] isolation level read committed [read write | read only];`
+3. `BEGIN [TRANSACTION]; SET TRANSACTION ISOLATION LEVEL READ COMMITTED;`
+4. `BEGIN [TRANSACTION]; SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL READ COMMITTED;`
+
+
+### 4 Examples
+
+```sql
+create table test (k int primary key, v int);
+```
+
+#### 1) Deadlocks are avoided in READ COMMITTED by relying on statement timeout.
+
+```sql
+truncate table test;
+insert into test values (1, 5);
+insert into test values (2, 5);
+```
+
+<table>
```

Comment:
Each table: headings for `Client 1` and `Client 2`.

Reply:
done